test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DependencyContextRazorViewEngineOptionsSetupTest.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Moq;
+using Xunit;
+using DependencyContextOptions = Microsoft.Extensions.DependencyModel.CompilationOptions;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public class DependencyContextRazorViewEngineOptionsSetupTest
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void GetCompilationOptions_ReturnsDefaultOptionsIfApplicationNameIsNullOrEmpty(string name)
+        {
+            // Arrange
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupGet(e => e.ApplicationName)
+                .Returns(name);
+            var setup = new TestableDependencyContextOptionsSetup(hostingEnvironment.Object);
+
+            // Act
+            var options = setup.GetCompilationOptionsPublic();
+
+            // Assert
+            Assert.Same(DependencyContextOptions.Default, options);
+        }
+
+        [Fact]
+        public void GetCompilationOptions_ReturnsDefaultOptionsIfApplicationDoesNotHaveDependencyContext()
+        {
+            // Arrange
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupGet(e => e.ApplicationName)
+                .Returns(typeof(Controller).GetTypeInfo().Assembly.GetName().Name);
+            var setup = new TestableDependencyContextOptionsSetup(hostingEnvironment.Object);
+
+            // Act
+            var options = setup.GetCompilationOptionsPublic();
+
+            // Assert
+            Assert.Same(DependencyContextOptions.Default, options);
+        }
+
+#if SOME_TEST_DEFINE
```

Comment:
Don't do this. Want the test to fail if someone removes the definition from the `project.json` file. (My suggestion was `#define` not `#if`.)